### PR TITLE
fix issue of detecting floating ips at KVM site

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

**Comment by zhenz** [comment]
@priteau The problem was the query. The original query has "INNER JOIN" with keystone.project to get project_name. It works fine at CHI sites, but KVM keystone.project is empty. Therefore, the query won't fail but will return nothing.

Please review the changes. Thanks!

**Comment by zhenz** [comment]
@priteau Looks like multiple problems in the code. Another problem is at CHI, the project_id in neutron.floatingips is always in lower case, no dash and using id, while at KVM, the project_id is actual project name. 

I'm done fixing that and tested on KVM server using dry-run. Please review again! Thanks!

File: hammers/query.py [hunk 1]
```diff
@@ -157,8 +163,8 @@ def owned_ip_single(db, project_id):
          , status
          , {projcol} AS project_id
     FROM   neutron.floatingips
-    WHERE  {projcol} = %s;
-    '''.format(projcol=project_col(db.version))
+    WHERE  {projcol} = {project_id};
+    '''.format(projcol=project_col(db.version), project_id=project_id)
```
Comment (priteau) [inline hammers/query.py]: Since project_id is set via args in the next line, you don't need to include it in format.

**Comment by zhenz** [comment]
@priteau 

The code only detects the ips/ports that are idled for certain days. And if some ips/ports are "active", the code will throw an exception. If you look at neutron_reaper line 116. I thought we want KVM to keep consistent on that with CHI. Do you want me to change it and only report the ones that are marked "DOWN"?

**Comment by priteau** [comment]
I think it's a general problem with the code, even for CHI. The function latest_instance_interaction() finds the last time an instance was created, updated, or deleted. It uses this timestamp to classify some projects as idle.

However, the reaper should also check that no instance is still running in the project to consider a project idle.

**Comment by zhenz** [comment]
@priteau Will fix accordingly. Thanks!

**Pull request merged**